### ISSUE TITLE
Close reason for PaginatorInterface

### ISSUE THREAD

**Issue by SnowyLuma** [Gorialis/jishaku#87]
## The Problem

I'd like to delete the invoking command when a user uses the close button on a PaginatorInterface.
As I do not use the `delete_message` option I don't want to unconditionally do this when the interface is done.

## The Ideal Solution

Have some way of knowing whether the interface was closed by the user, or was closed via the timeout.
```py
interface = PaginatorInterface(...)
await interface.send_to(channel)

await interface.task

if interface.closed_by_user:  # How exactly this looks doesn't really matter to me
    await ctx.message.delete()
```

## The Current Solution

One could wait for the message to be deleted, however this seems rather impractical if used in more than one location:
```py
interface = PaginatorInterface(...)
await interface.send_to(channel)

try:
    # This has the issue of not even working properly as the timeout counts from last button use
    await bot.wait_for('raw_message_delete', check=lambda x: x.message_id = interface.message.id, timeout=interface.timeout)
except asyncio.TimeoutError:
    return

await ctx.message.delete()
```

## Summary

Knowing whether a PaginatorInterface was closed manually or via the timeout.

**Comment by Gorialis** [comment]
This is implemented in 3a027993ac69b271d300e6ec6663c7b870b27641 (and thus will be part of v2) as `PaginatorInterface.close_exception`.

It should be now possible to do the following:
```py
if interface.closed:
    if interface.close_exception is None:
        # natural closure, e.g. by user pressing the close button
        ...
    elif isinstance(interface.close_exception, asyncio.TimeoutError):
        # closure through timeout
        ...
    elif isinstance(interface.close_exception, asyncio.CancelledError):
        # closure through cancellation, e.g. jsk cancel or the event loop closing
        ...
```

**Comment by SnowyLuma** [comment]
Thank you very much!
